### PR TITLE
Update SolveSetup.py

### PR DESCRIPTION
**Pull request by PipKat**
Not sure how this one PY file is used to separate the three separate pages under Setup in the API doc. Somehow it must though!

### CODE REVIEW

File: pyaedt/modules/SolveSetup.py [hunk 1]
```diff
@@ -141,27 +142,29 @@ def update(self, update_dictionary=None):
...
-            
-        intrinsics_list :
-            
-        isconvergence_list :
+        expressions_list : list
+            List of formulas to retrieve. 
+        report_type_list : list
+            List of report types for the expressions. 
+        intrinsics_list : list
+            List of instrinsic expressions for the expressions.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:156]:
```suggestion
            List of intrinsic expressions for the expressions. 
```

File: pyaedt/modules/SolveSetup.py [hunk 2]
```diff
@@ -230,27 +233,28 @@ def _expression_cache(self, expression_list, report_type_list,intrinsics_list, i
...
-        isrelativeconvergence :
-             (Default value = True)
-
+        expressions : str or list
+            One or more formulas to add to the expression cache. 
+        report_type : str or list, optional
+            Type of report for the expression. The default is ``Fields``. If a list of expressions 
+            is supplied, supply a corresponding list of report types.
+        intrinsics : str or list, optional
+            Instrinsic expression for the expression. The default is ``""``. If a list of expressions
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:246]:
```suggestion
            Intrinsic expression for the expression. The default is ``""``. If a list of expressions 
```

File: pyaedt/modules/SolveSetup.py [hunk 3]
```diff
@@ -580,26 +588,28 @@ def update(self, update_dictionary=None):
...
-            
-        isrelativeconvergence :
-            
-        conv_criteria :
+        expressions_list : list
+            List of formulas to retrieve. 
+        report_type_list : list
+            List of report types for the expressions. 
+        intrinsics_list : list
+            List of instrinsic expressions for the expressions.
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:600]:
```suggestion
            List of intrinsic expressions for the expressions. 
```

File: pyaedt/modules/SolveSetup.py [hunk 4]
```diff
@@ -668,28 +678,31 @@ def _expression_cache(self, expression_list, report_type_list, intrinsics_list,
...
-            Boolean if expression is in convergence criteria. It can be a list or a string (same length of expression_list) (Default value = True)
-        intrinsics :
-            List of Intrinsics of expression (if any). It can be a list or a string (same length of expression_list) (Default value = '')
+        expressions : str or list
+            One or more formulas to add to the expression cache. 
+        report_type : str or list, optional
+            Type of report for the expression. The default is ``Fields``. If a list of expressions 
+            is supplied, supply a corresponding list of report types.
+        intrinsics : str or list, optional
+            Instrinsic expression for the expression. The default is ``""``. If a list of expressions
```
Comment (PipKat) [inline pyaedt/modules/SolveSetup.py:691]:
```suggestion
            Intrinsic expression for the expression. The default is ``""``. If a list of expressions 
```

File: pyaedt/modules/SolveSetup.py [hunk 5]
```diff
@@ -141,27 +142,29 @@ def update(self, update_dictionary=None):
...
-            
-        intrinsics_list :
-            
-        isconvergence_list :
+        expressions_list : list
+            List of formulas to retrieve. 
+        report_type_list : list
+            List of report types for the expressions. 
+        intrinsics_list : list
+            List of intrinsic expressions for the expressions.
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:156]: expressions for the expressions?

File: pyaedt/modules/SolveSetup.py [hunk 6]
```diff
@@ -230,27 +233,28 @@ def _expression_cache(self, expression_list, report_type_list,intrinsics_list, i
...
+            Type of report for the expression. The default is ``Fields``. If a list of expressions 
+            is supplied, supply a corresponding list of report types.
+        intrinsics : str or list, optional
+            Intrinsic expression for the expression. The default is ``""``. If a list of expressions 
+            is supplied, supply a corresponding list of intrinsic expressesions.
+        isconvergence, bool, str, or list, optional
+            Whether the expression is in the convergence criteria. The  default is ``True``.  
+            If a list of expressions is supplied, supply a corresponding list of Boolean values.
+        isrelativeconvergence : 
+            The default is ``1``.
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:252]:
```suggestion
        isrelativeconvergence : bool
            The default is ``True``.
```

File: pyaedt/modules/SolveSetup.py [hunk 7]
```diff
@@ -364,23 +370,27 @@ def add_sweep(self, sweepname=None, sweeptype= "Interpolating"):
 
     @aedt_exception_handler
     def add_mesh_link(self, design_name, solution_name, parameters_dict, project_name="This Project*"):
-        """Add Mesh Link to another design. Design can be in the same project (default) or external project
-
+        """Add a mesh link to another design. 
+
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:374]:
```suggestion

```

File: pyaedt/modules/SolveSetup.py [hunk 8]
```diff
@@ -499,15 +507,12 @@ def __init__(self, parent,  solutiontype, setupname="MySetupAuto", isnewsetup=Tr
...
         
-        :return: the argument list
-
-        Parameters
-        ----------
-
         Returns
         -------
+        list
+           List of arguments.
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:515]:
```suggestion
            List of arguments.
```

File: pyaedt/modules/SolveSetup.py [hunk 9]
```diff
@@ -668,28 +678,31 @@ def _expression_cache(self, expression_list, report_type_list, intrinsics_list,
...
+        report_type : str or list, optional
+            Type of report for the expression. The default is ``Fields``. If a list of expressions 
+            is supplied, supply a corresponding list of report types.
+        intrinsics : str or list, optional
+            Intrinsic expression for the expression. The default is ``""``. If a list of expressions 
+            is supplied, supply a corresponding list of intrinsic expressesions.
+        isconvergence, bool, str, or list, optional
+            Whether the expression is in the convergence criteria. The  default is ``True``.  
+            If a list of expressions is supplied, supply a corresponding list of Boolean values.
         isrelativeconvergence :
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:696]:
```suggestion
        isrelativeconvergence : bool, optional
```

File: pyaedt/modules/SolveSetup.py [hunk 10]
```diff
@@ -668,28 +678,31 @@ def _expression_cache(self, expression_list, report_type_list, intrinsics_list,
...
-            List of Intrinsics of expression (if any). It can be a list or a string (same length of expression_list) (Default value = '')
+        expressions : str or list
+            One or more formulas to add to the expression cache. 
+        report_type : str or list, optional
+            Type of report for the expression. The default is ``Fields``. If a list of expressions 
+            is supplied, supply a corresponding list of report types.
+        intrinsics : str or list, optional
+            Intrinsic expression for the expression. The default is ``""``. If a list of expressions 
+            is supplied, supply a corresponding list of intrinsic expressesions.
+        isconvergence, bool, str, or list, optional
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:693]:
```suggestion
        isconvergence : bool, str, or list, optional
```

File: pyaedt/modules/SolveSetup.py [hunk 11]
```diff
@@ -668,28 +678,31 @@ def _expression_cache(self, expression_list, report_type_list, intrinsics_list,
...
-        report_type :
-            Report type of expression cache (eg. Fields). It can be a list or a string (same length of expression_list) (Default value = "Fields")
-        isconvergence :
-            Boolean if expression is in convergence criteria. It can be a list or a string (same length of expression_list) (Default value = True)
-        intrinsics :
-            List of Intrinsics of expression (if any). It can be a list or a string (same length of expression_list) (Default value = '')
+        expressions : str or list
+            One or more formulas to add to the expression cache. 
+        report_type : str or list, optional
+            Type of report for the expression. The default is ``Fields``. If a list of expressions
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:688]:
```suggestion
            Type of report for the expression. The default is ``"Fields"``. If a list of expressions 
```

File: pyaedt/modules/SolveSetup.py [hunk 12]
```diff
@@ -668,28 +678,31 @@ def _expression_cache(self, expression_list, report_type_list, intrinsics_list,
...
+            The default is ``1``.
+                
         Returns
         -------
-        type
-            none
-
+        bool
+            ``True`` when successful, ``False`` when failed.
+
```
Comment (akaszynski) [inline pyaedt/modules/SolveSetup.py:705]:
```suggestion

```

**Review by akaszynski — APPROVED** [review]
Approved with minor suggestions.

**Comment by akaszynski** [comment]
I'm starting to merge this as 15 active PRs seems like a lot and I'm sure the build machine wants something to do on the weekend.

**Pull request merged**